botnim/query.py

```diff
@@ -71,20 +73,31 @@ def _initialize_vector_store(self, config) -> VectorStoreES:
             production=is_production(self.environment),
         )
 
-    def search(self, query_text: str, num_results: int=None, explain: bool=False) -> List[SearchResult]:
+    def search(self, query_text: str, num_results: int=None, explain: bool=False, search_mode: Optional[SearchModeConfig] = None, mode: Optional[str] = None) -> List[SearchResult]:
         """
         Search the vector store with the given text
         
         Args:
             query_text (str): The text to search for
             num_results (int, optional): Number of results to return, or None to use context default
             explain (bool): Whether to include scoring explanation in results
+            search_mode (Optional[SearchModeConfig]): Optional search mode configuration
```

Comment:
better to implement as a single parameter and use `instanceof` in code to check actual type of parameter.